Source/EventFlow/Sagas/AggregateSagas/AggregateSaga.cs

```diff
@@ -68,11 +74,37 @@ public virtual async Task PublishAsync(ICommandBus commandBus, CancellationToken
         {
             var commandsToPublish = _unpublishedCommands.ToList();
             _unpublishedCommands.Clear();
+            var exceptions = new List<CommandException>();
 
             foreach (var unpublishedCommand in commandsToPublish)
             {
-                await unpublishedCommand(commandBus, cancellationToken).ConfigureAwait(false);
+                var executionResult = await unpublishedCommand.Item2(commandBus, cancellationToken).ConfigureAwait(false);
```

Comment:
@rasmus I have an idea to also catch exceptions from command execution and wrap them into CommandException so developer can just focus on CommandException in ISagaErrorHandler to know which command from saga failed. What do you think?

Reply:
I like the idea as it gives developers more information on which to act. The important this is to make sure that the exception doesn't have a reference to the command, aggregate or any other instance as this would really mess up IoC scopes

Reply:
You could pass the type of the command though, e.g. simply do a `command.GetType()`